packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -0,0 +1,186 @@
+    // value should be larger than the returned value.
+    function _getFundingRateAppliedTokenDebt(FixedPoint.Unsigned memory rawTokenDebt)
+        internal
+        view
+        returns (FixedPoint.Unsigned memory tokenDebt)
+    {
+        return rawTokenDebt.mul(cumulativeFundingRateMultiplier);
+    }
+
+    function _getFundingRateStore() internal view returns (FundingRateStoreInterface) {
+        return FundingRateStoreInterface(fpFinder.getImplementationAddress("FundingRateStore"));
+    }
+
+    // Fetches a funding rate from the Store, determines the period over which to compute an effective fee,
+    // and multiplies the current multiplier by the effective fee.
+    // A funding rate < 1 will reduce the multiplier, and a funding rate of > 1 will increase the multiplier.
+    // Note: 1 is set as the neutral rate because there are no negative numbers in FixedPoint, so we decide to treat
+    // values < 1 as "negative".
+    function _applyEffectiveFundingRatePerToken() internal {
+        uint256 currentTime = getCurrentTime();
+        uint256 paymentPeriod = currentTime.sub(lastUpdateTime);
+
+        FundingRateStoreInterface fundingRateStore = _getFundingRateStore();
+        FixedPoint.Unsigned memory _latestFundingRatePerSecondPerToken = fundingRateStore.getFundingRateForIdentifier(
+            identifier
+        );
+
+        FixedPoint.Unsigned memory effectiveFundingRateForPeriodPerToken;
+        (
+            cumulativeFundingRateMultiplier,
+            effectiveFundingRateForPeriodPerToken
+        ) = _calculateEffectiveFundingRatePerToken(
+            paymentPeriod,
+            _latestFundingRatePerSecondPerToken,
+            cumulativeFundingRateMultiplier
+        );
+        lastUpdateTime = currentTime;
+
+        emit NewFundingRate(
+            cumulativeFundingRateMultiplier.rawValue,
+            lastUpdateTime,
+            paymentPeriod,
+            _latestFundingRatePerSecondPerToken.rawValue,
+            effectiveFundingRateForPeriodPerToken.rawValue
+        );
+    }
+
+    function _calculateEffectiveFundingRatePerToken(
+        uint256 paymentPeriodSeconds,
+        FixedPoint.Unsigned memory fundingRatePerSecondPerToken,
+        FixedPoint.Unsigned memory feeMultiplier
+    ) internal pure returns (FixedPoint.Unsigned memory, FixedPoint.Unsigned memory) {
+        // Determine whether `fundingRatePerSecondPerToken` implies a negative or positive funding rate,
+        // and apply it over a pay period.
+        FixedPoint.Unsigned memory effectiveFundingRateForPeriodPerToken;
+        FixedPoint.Unsigned memory ONE = FixedPoint.fromUnscaledUint(1);
+
+        FixedPoint.Unsigned memory newFeeMultiplier;
+        if (fundingRatePerSecondPerToken.isEqual(ONE)) {
+            // If `fundingRatePerSecondPerToken` == 1, then maintain the current multiplier.
+
+            effectiveFundingRateForPeriodPerToken = ONE;
+            newFeeMultiplier = feeMultiplier;
```

Comment:
nit: could we just remove this case entirely and have this still work as expected?

Reply:
Yep, fixed!